OracleSOASuite/dockerfiles/12.2.1.3/README.md

```diff
@@ -0,0 +1,170 @@
+
+## Creating a container for Administration Server
+
+Start a container to launch the Administration Server from the image created using above steps. The environment variables used to configure the domain are defined in `adminserver.env.list` file. Replace in `adminserver.env.list` the values for the Database and WebLogic passwords. 
+
+Create an environment file adminserver.env.list
+
+    CONNECTION_STRING=<Database Container Name>:<port#>/<ORACLE_PDB>
+    RCUPREFIX=<RCU_Prefix>
+    DB_PASSWORD=<database_sys_password>
+    DB_SCHEMA_PASSWORD=<soa-infra schema password>
+    ADMIN_PASSWORD=<admin_password>
+    MANAGED_SERVER=<Managed Server name>
+    DOMAIN_TYPE=<soa/osb/soaosb/soaess/soaessosb/bpm>
+    
+Sample Data will look like this:
+
+    CONNECTION_STRING=soadb:1521/soapdb
+    RCUPREFIX=SOA1
+    DB_PASSWORD=Welcome1
+    DB_SCHEMA_PASSWORD=Welcome1
+    ADMIN_PASSWORD=Welcome1
+    MANAGED_SERVER=soa_server1
+    DOMAIN_TYPE=soa
+    
+To start a docker container with a SOA domain and the WebLogic Administration Server call `docker run` command and pass the above `adminserver.env.list` file.
+
+A sample docker run command is given below:
+
+     $docker run -i -t  --name soaas --network=SOANet -p 7001:7001  -v /scratch/DockerVolume/SOAVolume/SOA:/u01/oracle/user_projects   --env-file ./adminserver.env.list oracle/soa:12.2.1.3.0
+     
+>IMPORTANT: the resulting images will NOT have a domain pre-configured. But, it has the scripts to create and configure a soa domain >while creating a container out of the image.
+     
+The options `-i -t` in the above command runs the container in interactive mode and you will be able to see the commands running in the container. This includes the command for RCU creation, domain creation and configuration followed by starting the Administration Server. 
+
+>IMPORTANT: You need to wait till all the above commands are run before you can access the Administration Server Web Console.
+>The following lines highlight when the Administration Server is ready to be used:    
+
+`INFO: Admin server is running`
+
+`INFO: Admin server running, ready to start managed server`
+
+The above line indicate that the Administration Server started successfully with the name `soaas`, mapping container port 7001 to host port 7001 enables accessing of the Weblogic host outside of the local host, connecting to SOANet network enables accessing the DB container by it's name i.e soadb. This includes the command to tail logs to keep the container up and running.
+
+
+## Creating a container for SOA Server
+
+Start a container to launch the Managed Server from the image created. The environment variables used to run the Managed Server image are defined in the file `soaserver.env.list`. 
+
+Create an environment file `soaserver.env.list` with the below details:
+
+    MANAGED_SERVER=<Managed Server name, For Exp:- soa_server1, osb_server1, bpm_server1>
+    DOMAIN_TYPE=<soa/osb/bpm>
```

Comment:
What's the difference between the three? Which one should I pick? What do they mean?

Reply:
These are 3 different applications running as containers from the same image. This is runtime setting not a build time setting. You just create one single SOA image and users can start different containers based on this environment settings. Hope this clarifies.

Reply:
This clarification should be in the README.  I had the question about how to create a OSB domain.